server/plugin.go

```diff
@@ -1,304 +0,0 @@
-	}
-	urlLine.RawQuery = parametersURLLine.Encode()
-	line := "![](" + urlLine.String() + ") "
-
-	m = m + pie + bar + line
-
-	return &model.CommandResponse{
-		ResponseType: model.COMMAND_RESPONSE_TYPE_IN_CHANNEL,
-		Text:         m,
-	}, nil
-
-}
-
-type analyticsData struct {
-	displayName string
-	name        string
-	nb          int64
-	reply       int64
-}
-
-type preparedData struct {
-	totalMessagesPublic  int64
-	totalMessagesPrivate int64
-	users                []analyticsData
-	channels             []analyticsData
-}
-
-func (p *Plugin) prepareData() (*preparedData, *model.AppError) {
-	p.currentAnalytic.RLock()
-	defer p.currentAnalytic.RUnlock()
-
-	totalMessagesPublic := int64(0)
-	totalMessagesPrivate := int64(0)
-	users := make([]analyticsData, 0)
-	channels := make([]analyticsData, 0)
-	channels = append(channels, analyticsData{"Private", "Private", 0, 0})
-
-	for key, nb := range p.currentAnalytic.Channels {
-		channelName, channelDisplayName, err := p.getChannelName(key)
-		if err != nil {
-			return nil, err
-		}
-		if channelName == "Private" {
-			totalMessagesPrivate += nb
-			channels[0].nb = channels[0].nb + nb
-		} else {
-			totalMessagesPublic += nb
-			channels = p.updateOrAppend(channels, analyticsData{channelDisplayName, channelName, nb, 0})
-		}
-	}
-	for key, nb := range p.currentAnalytic.ChannelsReply {
-		channelName, channelDisplayName, err := p.getChannelName(key)
-		if err != nil {
-			return nil, err
-		}
-		channels = p.updateOrAppend(channels, analyticsData{channelDisplayName, channelName, 0, nb})
-	}
-	for key, nb := range p.currentAnalytic.Users {
-		displayKey, err := p.getUsername(key)
-		if err != nil {
-			return nil, err
-		}
-		users = p.updateOrAppend(users, analyticsData{displayKey, displayKey, nb, 0})
-	}
-	for key, nb := range p.currentAnalytic.UsersReply {
-		displayKey, err := p.getUsername(key)
-		if err != nil {
-			return nil, err
-		}
-		users = p.updateOrAppend(users, analyticsData{displayKey, displayKey, 0, nb})
-	}
-	sort.Slice(users, func(i, j int) bool {
-		return users[i].nb > users[j].nb
-	})
-	sort.Slice(channels, func(i, j int) bool {
-		return channels[i].nb > channels[j].nb
-	})
-	return &preparedData{
-		totalMessagesPublic:  totalMessagesPublic,
-		totalMessagesPrivate: totalMessagesPrivate,
-		users:                users,
-		channels:             channels,
-	}, nil
-}
-
-func (p *Plugin) updateOrAppend(originals []analyticsData, upsert analyticsData) []analyticsData {
-	var original analyticsData
-	for index, value := range originals {
-		if value.name == upsert.name {
-			original = value
-			originals = append(originals[:index], originals[index+1:]...)
-			break
-		}
-	}
-	if original.name != "" {
-		if upsert.nb == 0 {
-			upsert.nb = original.nb
-		}
-		if upsert.reply == 0 {
-			upsert.reply = original.reply
-		}
-	}
-	return append(originals, upsert)
-}
-
-func (p *Plugin) getChannelName(key string) (string, string, *model.AppError) {
```

Comment:
1/5 that you should return `error` instead of `*model.AppError` here.

Reply:
:heavy_check_mark: